apps/internal/oauth/ops/authority/authority.go

```diff
@@ -98,6 +98,46 @@ func (r *TenantDiscoveryResponse) Validate() error {
 	return nil
 }
 
+// ValidateIssuerMatchesAuthority validates that the issuer in the TenantDiscoveryResponse matches the authority.
+// This is used to prevent token issuers from impersonating legitimate authorities.
+func (r *TenantDiscoveryResponse) ValidateIssuerMatchesAuthority(authorityURI string) error {
+	if authorityURI == "" {
+		return nil
+	}
+
+	// Parse the issuer URL
+	issuerURL, err := url.Parse(r.Issuer)
+	if err != nil {
+		return fmt.Errorf("TenantDiscoveryResponse: failed to parse issuer URL: %w", err)
+	}
+
+	// Even if it doesn't match the authority, issuers from known and trusted Microsoft hosts are valid
+	if TrustedHost(issuerURL.Host) {
+		return nil
+	}
+
+	// Check against other trusted Microsoft hosts
+	switch issuerURL.Host {
+	case loginMicrosoft, loginWindows, loginSTSWindows, defaultHost:
```

Comment:
There are a bunch of aliases on soverigns too

Reply:
The `openIDConfigurationEndpoint` in this `resp` response contains metadata and all the aliases, we can check this through issuer validation.

Reply:
@4gust : It looks like the `InstanceDiscoveryResponse` has the alias metadata, so are you talking about the `resp` down in the `AADInstanceDiscovery` function? How would I access the `InstanceDiscoveryResponse` from the new validation function?

Reply:
You can add a new Aliases field to the cache struct inside authorityEndpoint. Include corresponding setter and getter methods.

These methods should update the AADInstanceDiscovery within the openIDConfigurationEndpoint.

This way, aliases can be managed per authorityEndpoint and accessed externally as needed, while keeping the logic scoped and consistent.

Reply:
> There are a bunch of aliases on soverigns too
@bgavrilMS : Forgot to respond to this before the newest changes, but the sovereign clouds were covered by a different check a few lines above (`if TrustedHost(issuerURL.Host)`)

However, as of the latest commit that check was removed and replaced with a check on the actual aliases returned from  instance discovery.